prestashop1.6/upgrade/Upgrade-1.2.0.php

```diff
@@ -0,0 +1,139 @@
+<?php
+/**
+ * tawk.to
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://opensource.org/licenses/osl-3.0.php
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to support@tawk.to so we can send you a copy immediately.
+ *
+ * @author tawkto support@tawk.to
+ * @copyright Copyright (c) 2014-2021 tawk.to
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+
+if (!defined('_PS_VERSION_')) {
+    exit;
+}
+
+function upgrade_module_1_2_0()
+{
+    $db = Db::getInstance();
+
+    // remove the suffix first on the config key names.
+    $remove_suffix_result = remove_suffix($db);
+    if (!$remove_suffix_result) {
+        return false;
+    }
+
+    // force reload cache
+    Configuration::loadConfiguration();
+
+    // insert the new records for TAWKTO_WIDGET_CURRENT.
+    $insert_records_result = insert_records();
+    if (!$insert_records_result) {
+        return false;
+    }
+
+    // delete records for TAWKTO_WIDGET_PAGE_ID & TAWKTO_WIDGET_WIDGET_ID
+    $remove_extras_result = remove_extras();
+    if (!$remove_extras_result) {
+        return false;
+    }
+
+    return true;
+}
+
+function remove_suffix($db)
+{
+    $keys = array(
+        TawkTo::TAWKTO_WIDGET_PAGE_ID,
+        TawkTo::TAWKTO_WIDGET_WIDGET_ID,
+        TawkTo::TAWKTO_WIDGET_OPTS,
+        TawkTo::TAWKTO_WIDGET_USER
+    );
+
+    // start building the update sql statement
+    $sql = array(
+        'UPDATE '._DB_PREFIX_.bqSQL(Configuration::$definition['table']).' conf',
+        'SET conf.name = CASE'
+    );
+
+    // build case when clause
+    foreach ($keys as $key) {
+        array_push($sql, "WHEN conf.name LIKE '".pSQL($key)."%' THEN '".pSQL($key)."'");
+    }
+
+    // build else and end case clause and where clause
+    array_push(
+        $sql,
+        "ELSE conf.name",
+        "END",
+        "WHERE conf.name LIKE 'TAWKTO_WIDGET_%';"
+    );
+
+    // join sql array and execute
+    $result = $db->execute(join(' ', $sql)); //returns boolean value
+
+    return $result;
+}
+
+function insert_records()
+{
+    $res = true;
+
+    // modify global first
+    $res &= modify_widget();
+
+    $shop_ids = Shop::getCompleteListOfShopsID();
+
+    $updated_groups = array();
+    foreach ($shop_ids as $shop_id) {
```

Comment:
Saw the approach of this from one of the plugins developed by Prestashop.

https://github.com/PrestaShop/homeslider/blob/master/upgrade/install-1.5.0.php.

I used a different function to get the `$shop_ids` since the function they used to get those had problems when the plugin was updated while a specific store was selected instead of `All Shops`.